docs/io/ascii/fast_ascii_io.rst

```diff
@@ -35,21 +37,24 @@ To disable the fast engine, specify ``fast_reader=False`` or
 
 .. Note:: Guessing and Fast reading
 
-   By default |read| will try to guess the format of in the input data by successively
-   trying different formats until one succeeds ([reference the guessing section]).
-   For the default ``'ascii'`` format this means that a number of pure Python readers
-   with no fast implementation will be tried before getting to the fast readers.
+   By default |read| will try to guess the format of in the input data by
+   successively trying different formats until one succeeds
+   (see the section on :ref:`guess_formats`).
+   For the default ``'ascii'`` format it will try all fast reader formats
```

Comment:
As mentioned in the first comment, I don't think this is the case, and in fact it is crucial to maintain the guess order.

Reply:
Agreed, this is not the default, and in the case of a explicitly required `fast_reader` the slow readers will/should not be tested at all. The previous version omitted this latter point, but this should be worded more clearly.